velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -70,17 +81,8 @@ std::unique_ptr<velox::dwrf::Writer> HiveDataSink::createWriter() {
   // Without explicitly setting flush policy, the default memory based flush
   // policy is used.
 
-  auto hiveWriterParameters =
-      std::dynamic_pointer_cast<const HiveWriterParameters>(
-          writeProtocol_->getWriterParameters(
-              insertTableHandle_, connectorQueryCtx_));
-  VELOX_CHECK_NOT_NULL(
-      hiveWriterParameters,
-      "Hive data sink expects write parameters for Hive.");
-  writerParameters_.emplace_back(hiveWriterParameters);
-
-  auto writePath = fs::path(hiveWriterParameters->writeDirectory()) /
-      hiveWriterParameters->writeFileName();
+  auto writePath = fs::path(writerParameters->writeDirectory()) /
```

Comment:
This change looks like a code movement without any logic change. Is this so? Why is it needed?

Reply:
Yeah, it's just a code movement. I'm moving the `auto hiveWriterParameters = writeProtocol_->getWriterParameters()` out of this createWriter() class and to append(), and pass hiveWriterParameters to createWriter(WriterParameters). It's because in the next PR we need to pass partitionDirectory when calling writeProtocol_->getWriterParameters(). I feel it's more clean to pass WriterParameters to createWriter() compared w/ passing partitionDirectory.